.github/workflows/k8s_matrix_test.yaml

```
@@ -0,0 +1,14 @@
+# Runs the single-cluster e2e tests with various versions of Kubernetes, to ensure backward compatibility.
+name: K8s version test
+on:
+  schedule:
+    - cron: '0 12 * * 0' # Every Sunday at noon UTC
+jobs:
+  run-e2e-tests:
+    strategy:
+      fail-fast: false
+      matrix:
+        kind_node_version: ["v1.22.15", "v1.23.6", "v1.24.7"]
```

We should probably test all the version we're supposed to support.

```suggestion
        kind_node_version: ["v1.21.14", "v1.22.15", "v1.23.6", "v1.24.7"]
```

---

1.25 and 1.26?

---

Right, I was going to ask about which versions to cover.
1.25 is used in the regular PR actions, maybe we can leave it out.

---

We could put 1.25 for completeness and avoiding having to update this workflow once we move our base tests to 1.26.
I've approved the PR nonetheless, so you can add 1.25 and then merge it.